src/PrestaShopBundle/Controller/Admin/Improve/Design/ImageSettingsController.php

```diff
@@ -0,0 +1,315 @@
+use PrestaShop\PrestaShop\Core\Domain\ImageSettings\Command\RegenerateThumbnailsCommand;
+use PrestaShop\PrestaShop\Core\Domain\ImageSettings\Exception\ImageTypeNotFoundException;
+use PrestaShop\PrestaShop\Core\Domain\ImageSettings\Query\GetImageTypeForEditing;
+use PrestaShop\PrestaShop\Core\Domain\ImageSettings\QueryResult\EditableImageType;
+use PrestaShop\PrestaShop\Core\Search\Filters\ImageTypeFilters;
+use PrestaShopBundle\Controller\Admin\FrameworkBundleAdminController;
+use PrestaShopBundle\Form\Admin\Improve\Design\ImageSettings\DeleteImageTypeType;
+use PrestaShopBundle\Form\Admin\Improve\Design\ImageSettings\RegenerateThumbnailsType;
+use PrestaShopBundle\Security\Annotation\AdminSecurity;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+
+/**
+ * Responsible for Image Settings actions in Back Office
+ */
+class ImageSettingsController extends FrameworkBundleAdminController
+{
+    /**
+     * Displays image settings listing page.
+     *
+     * @AdminSecurity("is_granted('read', request.get('_legacy_controller'))")
+     *
+     * @param Request $request
+     * @param ImageTypeFilters $filters
+     *
+     * @return Response
+     */
+    public function indexAction(Request $request, ImageTypeFilters $filters): Response
+    {
+        // Get image type grid
+        $imageTypeGridFactory = $this->get('prestashop.core.grid.factory.image_type');
+        $imageTypeGrid = $imageTypeGridFactory->getGrid($filters);
+
+        // Create form for deleting image type if needed (in modal)
+        $deleteForm = $this->createForm(DeleteImageTypeType::class);
+
+        // Create form to set some image settings
+        $configFormBuilder = $this->get('prestashop.admin.image_settings.form_handler');
+        $configForm = $configFormBuilder->getForm();
+
+        // Create form to regenerate thumbnails
+        $regenThumbnailsForm = $this->createForm(RegenerateThumbnailsType::class);
+
+        return $this->render('@PrestaShop/Admin/Improve/Design/ImageSettings/index.html.twig', [
+            'help_link' => $this->generateSidebarLink($request->attributes->get('_legacy_controller')),
+            'imageTypeGrid' => $this->presentGrid($imageTypeGrid),
+            'enableSidebar' => true,
+            'deleteImageTypeForm' => $deleteForm->createView(),
+            'layoutHeaderToolbarBtn' => [
+                'add' => [
+                    'href' => $this->generateUrl('admin_image_settings_create'),
+                    'desc' => $this->trans('Add new image type', 'Admin.Design.Feature'),
+                    'icon' => 'add_circle_outline',
+                ],
+            ],
+            'configForm' => $configForm->createView(),
+            'regenThumbnailsForm' => $regenThumbnailsForm->createView(),
+        ]);
+    }
+
+    public function saveSettingsAction(Request $request): Response
```

Comment:
Isn't there an `AdminSecurity` attribute/annotation missing here? We should check permissions on such a controller This one is blocking IMHO